store/store.go

```diff
@@ -308,22 +305,23 @@ func (s *Store[H]) HasAt(_ context.Context, height uint64) bool {
 
 // DeleteTo implements [header.Store] interface.
 func (s *Store[H]) DeleteTo(ctx context.Context, to uint64) error {
-	var from uint64
-
-	if tailPtr := s.tailHeader.Load(); tailPtr != nil {
-		from = (*tailPtr).Height()
+	head, err := s.Head(ctx)
+	if err != nil {
+		return fmt.Errorf("header/store: reading head: %w", err)
 	}
-	if from >= to {
-		log.Debugf("header/store: attempt to delete empty range(%d, %d)", from, to)
```

Comment:
Decided to revert idempotency. In this case its more valuable to error the user(us) if we passed wrong `to` height, instead of silently succeeding.